### ISSUE TITLE
Inconsistent icon size with the rest of the panel icons

### ISSUE THREAD

**Issue by madjoe** [Zren/plasma-applet-todolist#71]
The width of the icon should be a bit wider, or equal in size to the rest of the icons on my plasma panel.
The height of the panel is 61 for this particular example:

![](https://i.imgur.com/NC0Emun.png)

**Comment by Zren** [comment]
I don't see a reason why it's a different size.

Perhaps try changing the todolist icon to something else?

* https://github.com/KDE/plasma-desktop/blob/master/desktoppackage/contents/applet/DefaultCompactRepresentation.qml
* https://github.com/Zren/plasma-applet-todolist/blob/master/package/contents/ui/main.qml

**Comment by madjoe** [comment]
> Perhaps try changing the todolist icon to something else?

I just tried this and unfortunately, no matter which Icon I choose, it always remains tiny as the default one.

**Comment by Zren** [comment]
Can I see a screenshot of a custom icon? All I'm seeing is a semi transparent rectangle on the right side in your first shortcut. It looks like the widget is taking up an 64x64 area, just like the app launchers.

The round blue counter is supposed to only cover a quarter of the icon area. Pretty sure you can adjust the size in the settings.

**Comment by madjoe** [comment]
![](https://i.imgur.com/fjhK6ue.png)

![](https://i.imgur.com/vu9J7mo.png)

**Comment by Zren** [comment]
Latte Dock?

**Comment by madjoe** [comment]
> Latte Dock?

No, it's the default Plasma panel.

![](https://i.imgur.com/lfyvVyz.png)

> Edit: Also, what happens why you enable "Use big counter"?

I didn't enable "Use big counter". The only counter style enabled is "Use round counter".

**Comment by Zren** [comment]
Bah typo.  What happens when* you enable "Use big counter"?

**Comment by madjoe** [comment]
![](https://i.imgur.com/WXa2zBP.png)

**Comment by Zren** [comment]
So it's detecting the height fine, but the width... ah, it's probably because it's a "dock like panel" which expands. Since the compact view probably doesn't have a minimum width, it's being squished.

**Comment by madjoe** [comment]
Not really, no. I expanded the panel so it has plenty of room and it still looks squished.

![](https://i.imgur.com/x7F15gF.png)

**Comment by Zren** [comment]
Yeah, that doesn't look like the problem. Have you restarted plasma since adding the widget to the panel?

![2021-03-25___21-53-24](https://user-images.githubusercontent.com/416367/112565971-87fbfc00-8db4-11eb-8c03-e06fbf568a30.png)

**Comment by madjoe** [comment]
```
qdbus org.kde.KWin /KWin reconfigure; kquitapp5 plasmashell && kstart5 plasmashell > /dev/null 2>&1
```
Nothing has changed... for testing purposes I increased the height a bit:
![](https://i.imgur.com/6SMhF6g.png)

**Comment by Zren** [comment]
What's the icon size in System Settings > Icons > Configure Icon Sizes?

![2021-03-26___00-02-31](https://user-images.githubusercontent.com/416367/112575373-94895000-8dc6-11eb-95d1-3a818419917e.png)

The only thing I can think of is that there's a binding loop.

```
journalctl -b0 _COMM=plasmashell | grep todolist
```

**Comment by madjoe** [comment]
> What's the icon size in System Settings > Icons > Configure Icon Sizes?
> 
> ![2021-03-26___00-02-31](https://user-images.githubusercontent.com/416367/112575373-94895000-8dc6-11eb-95d1-3a818419917e.png)

It's exactly the same as per your screenshot: 48.

> The only thing I can think of is that there's a binding loop.
> 
> ```
> journalctl -b0 _COMM=plasmashell | grep todolist
> ```

```
$ journalctl -b0 _COMM=plasmashell | grep todolist
Mar 26 08:14:27 mypc plasmashell[1664]: file:///home/me/.local/share/plasma/plasmoids/com.github.zren.todolist/contents/ui/NoteListView.qml:19:12: QML TodoItemDelegate: Binding loop detected for property "implicitHeight"
Mar 26 08:14:27 mypc plasmashell[1664]: file:///home/me/.local/share/plasma/plasmoids/com.github.zren.todolist/contents/ui/NoteListView.qml:19:12: QML TodoItemDelegate: Binding loop detected for property "implicitHeight"
Mar 26 08:14:27 mypc plasmashell[1664]: file:///home/me/.local/share/plasma/plasmoids/com.github.zren.todolist/contents/ui/NoteListView.qml:19:12: QML TodoItemDelegate: Binding loop detected for property "implicitHeight"
```

**Comment by Zren** [comment]
> NoteListView.qml:19:12: QML TodoItemDelegate: Binding loop detected for property "implicitHeight"

That's from the popup/full view, not the panel/compact view. It's caused by the todo item TextFields resizing their height after formatting links. It's "normal".

I'm out of ideas, sorry.

**Comment by Zren** [comment]
Wait, I forgot your panel is like 61px tall. If they're limited to 48px, then they can't grow bigger. Try changing the panel icon to 64px.

**Comment by madjoe** [comment]
> Wait, I forgot your panel is like 61px tall. If they're limited to 48px, then they can't grow bigger. Try changing the panel icon to 64px.

As you can see from your screenshot above, 48px is the very maximum value you could assign in the panel icon size setting. I tried to reduce the height of my panel to 48px (below), but still the icon height is not consistent with the rest of the icons.

![](https://i.imgur.com/QgblIVV.png)

> Also, which widget is to the right of the todolist? It looks like the icon is larger than the "task manager" widget icons.

The widget is [Sticky Note](https://github.com/KDE/kdeplasma-addons/blob/master/applets/notes/package/metadata.desktop) by authors Davide Bettio and Lukas Kropatschek. It gets significantly larger only when I increase the height of the panel to >68px.

**Comment by Zren** [comment]
Sticky Notes doesn't seem to follow the configured max panel icon size.

* https://github.com/KDE/kdeplasma-addons/blob/master/applets/notes/package/contents/ui/main.qml#L94

You can try removing the line with `Layout.maximumWidth` then restarting Plasma.

* https://github.com/Zren/plasma-applet-todolist/blob/master/package/contents/ui/main.qml#L43-L44
* `~/.local/share/plasma/plasmoids/com.github.zren.todolist/contents/ui/main.qml`

**Comment by madjoe** [comment]
> You can try removing the lines with `Layout.maximum____` then restarting Plasma.

This workaround did the trick. Now I can see a normal-size icon as it should be by the default:
![](https://i.imgur.com/y45XH7F.png)

Can you deploy it as the default state (deleting those two `Layout.maximum____` lines)?

> You can also try deleting `IconCounterOverlay { ... }` section.
This step was unnecessary, since it just hid the counter. The icon size was still normal though.

**Comment by madjoe** [comment]
With the recent change (removing the lines with `Layout.maximum____`) I've also noticed the delete icon became so tiny.

![](https://i.imgur.com/nj8WiOW.png)

Not sure if it was intended or not, but it seems too small.

**Comment by iamkroot** [comment]
@madjoe I was able to fix it by editing the following lines to scale by a factor of 2. 
https://github.com/Zren/plasma-applet-todolist/blob/b537147389ab0031544089b85a77e135b7dadf37/package/contents/ui/TodoItemDelegate.qml#L259-L260

becomes
```qml
Layout.preferredWidth: checkbox.implicitHeight * 2
Layout.preferredHeight: checkbox.implicitHeight * 2
```
